lib/quick_spreadsheet.rb

```diff
@@ -3,50 +3,92 @@
 
 class QuickSpreadsheet
   def self.args
-    puts "title: string"
-    puts "header_row: Array<string>"
-    puts "body_rows: Array<Array<string>>"
+    puts "filename: string (optional)"
     puts "folder_path: string (optional)"
+    puts "Variant 1:"
+    puts "  title: string"
+    puts "  header_row: Array<string>"
+    puts "  body_rows: Array<Array<string>>"
+    puts "Variant 2:"
+    puts "  sheets: Array<Array<"
+    puts "    title: string"
+    puts "    header_row: Array<string>"
+    puts "    body_rows: Array<Array<string>>"
   end
 
   def self.call(args)
-    title       = args[:title]
-    header_row  = args[:header_row]
-    body_rows   = args[:body_rows]
-    folder_path = args[:folder_path] || "#{Dir.pwd}/tmp"
-    format      = :xls
+    time = Time.now
+    file_timestamp = time.strftime("%Y-%m-%d-%H.%M")
 
+    folder_path = args[:folder_path] || "#{Dir.pwd}/tmp"
     FileUtils::mkdir_p(folder_path) unless File.exists?(folder_path)
+    format      = :xls
 
-    time = Time.now
-    file_timestamp = time.strftime("%Y-%m-%d-%H.%M")
-    filename = "#{title.gsub(" ", "")}_g#{file_timestamp}.#{format.to_s}"
+    sheets = if args[:sheets]
+      args[:sheets]
+    else
+      [{
+        title:      args[:title],
+        header_row: args[:header_row],
+        body_rows:  args[:body_rows]
+      }]
+    end
+    
+    filename = [ args[:filename], format.to_s ].join(".") or
+      "#{sheets[0][:title].gsub(" ", "")}_g#{file_timestamp}.#{format.to_s}"
```

Comment:
- are you sure this does what you think it does?  i don't think the second clause will ever be evaluated, since:
`[nil, "xls"].join(".")` => `".xls"`
`".xls" or "whatever"` => `".xls"`

- did you test that this branch works with the original one-sheet api?

- also in general I think `or` should be avoided, since it doesn't have the same behaviour as `||` and I don't think many ruby devs know about the differences between the two having to do with operator precedence (I certainly don't)

what you probably want is:
```
filename =
  if args[:filename].nil?
    "#{sheets[0][:title].gsub(" ", "")}_g#{file_timestamp}.#{format.to_s}"
  else
    [ args[:filename], format.to_s ].join(".") 
  end   
```